logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -1,48 +1,42 @@
 package com.wire.kalium.logic
 
-import com.wire.kalium.cryptography.ProteusClient
 import com.wire.kalium.logic.configuration.ClientConfig
-import com.wire.kalium.logic.feature.UserSessionScope
+import com.wire.kalium.logic.data.session.SessionMapper
+import com.wire.kalium.logic.data.session.SessionMapperImpl
+import com.wire.kalium.logic.feature.UserSessionScopeCommon
 import com.wire.kalium.logic.feature.auth.AuthSession
-import com.wire.kalium.logic.feature.auth.AuthSessionMapper
 import com.wire.kalium.logic.feature.auth.AuthenticationScope
-import com.wire.kalium.network.AuthenticatedNetworkContainer
 import com.wire.kalium.network.LoginNetworkContainer
 
-class CoreLogic(
-    private val clientLabel: String,
-    private val rootProteusDirectoryPath: String,
-    private val authSessionMapper: AuthSessionMapper = AuthSessionMapper(),
-    private val clientConfig: ClientConfig
+expect class CoreLogic: CoreLogicCommon
+
+abstract class CoreLogicCommon(
+    // TODO: can client label be replaced with clientConfig.deviceName() ?
+    protected val clientLabel: String,
+    protected val rootProteusDirectoryPath: String,
 ) {
 
-    private val loginContainer: LoginNetworkContainer by lazy {
+    protected val loginContainer: LoginNetworkContainer by lazy {
         LoginNetworkContainer()
     }
 
-    private val userScopeStorage = hashMapOf<AuthSession, AuthenticatedDataSourceSet>()
+    protected val userScopeStorage = hashMapOf<AuthSession, AuthenticatedDataSourceSet>()
     // TODO: - Update UserSession when token is refreshed
     //       - Delete UserSession and DataSourceSets when user logs-out
 
     @Suppress("MemberVisibilityCanBePrivate") // Can be used by other targets like iOS and JS
-    fun getAuthenticationScope(): AuthenticationScope = AuthenticationScope(loginContainer, clientLabel)
+    abstract fun getAuthenticationScope(): AuthenticationScope
+
+    protected abstract val clientConfig: ClientConfig
+
+    protected val sessionMapper: SessionMapper get() = SessionMapperImpl()
 
     @Suppress("MemberVisibilityCanBePrivate") // Can be used by other targets like iOS and JS
-    fun getSessionScope(session: AuthSession): UserSessionScope {
-        val dataSourceSet = userScopeStorage[session] ?: run {
-            val networkContainer = AuthenticatedNetworkContainer(authSessionMapper.toSessionCredentials(session))
-            val proteusClient = ProteusClient(rootProteusDirectoryPath, session.userId)
-            AuthenticatedDataSourceSet(networkContainer, proteusClient).also {
-                userScopeStorage[session] = it
-            }
-        }
-        return UserSessionScope(session, dataSourceSet, clientConfig)
-    }
+    abstract fun getSessionScope(session: AuthSession): UserSessionScopeCommon
```

Comment:
Would we need to make this whole method abstract if we had already the `ClientConfig` value injected?

Reply:
`UserSessionScope` in android need `Context`

Reply:
Not really, we only need the `Context` to populate the `ClientConfig` class, so we could pass an instance of this as a parameter instead no?

Reply:
Would another alternative be to introduce a factory which contains all the platform specific parts

```
// android
actual CoreLogicFactory(private applicationContext: Context) {
	
}

expect CoreLogicFactory {

	val clientConfig: ClientConfig
	val databaseDriverFactory: DatabaseDriverFactory
	val coreLogic: CoreLogic
}

CoreLogic(clientConfig: ClientConfig, databaseDriverFactory) {
	
}
```